x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/exchange/ExchangeService.java

```diff
@@ -172,6 +173,24 @@ public static void openExchange(
         );
     }
 
+    public void addExchangeSourceHandler(String sessionId, ExchangeSourceHandler sourceHandler) {
+        exchangeSources.put(sessionId, sourceHandler);
+    }
+
+    public ExchangeSourceHandler removeExchangeSourceHandler(String sessionId) {
+        return exchangeSources.remove(sessionId);
+    }
+
+    public void finishSessionEarly(String sessionId, ActionListener<Void> listener) {
```

Comment:
Where is this method called - only on the coordinator or possibly also on data nodes or remote clusters running parts of the ES|QL query?